src/str_utils.rs

```diff
@@ -59,6 +59,25 @@ pub(crate) fn utf16_code_unit_to_char_idx(text: &str, utf16_idx: usize) -> usize
     str_indices::chars::from_byte_idx(text, str_indices::utf16::to_byte_idx(text, utf16_idx))
 }
 
+/// Returns the byte position just after the fist line break
+/// in `text`, or `None` if there is no line break.
+/// This function currrently does not utilise SIMD
+pub fn next_line_break(text: &str) -> Option<usize> {
```

Comment:
You should be able to implement this in terms of `lines::to_byte_idx()`, I think.  Basically just `to_byte_idx(text, 1)` with an additional check if you get a one-past-the-end result.

Your implementation here is also incorrect with respect to feature flags: the recognized line endings are configurable via feature flag, whereas your implementation here is fixed to always recognize all line endings regardless of feature flags.  So implementing in terms of `lines::to_byte_idx()` would fix that as well (the correct `lines` module is conditionally imported in this file already).

Reply:
The difference is a lot more subtle, I used to actually do exactly that but it caused subtly broken results, because you can not differentiate between a line terminator at the end of the line and no line terminator. For example `to_byte_idx("fo\n", 1)` and  `to_byte_idx("foo", 1)` both output  `3`. Correctly handling this would require checking for a line ending again which is really cumbersome and also slow (remeber that this occurs at the end of every chunk not just EOF). Furthermore the case of only looking for the first newline terminator can be easier optimized because it requires fewer checks so its even faster for small lines (which are fairly common). I will try to put together a draft PR to `str_indicies` tomorrow I sadly had a bit much on my plate today. I would also like to SIMD accelerate the reverse case which is not possible to implement with any function is `str_indicies` at all (and was already custom implemented for the old lines iterator). It won't be a big PR at all (like 30 lines for the forward case) because the core SIMD function for counting lines can be reused.

Reply:
> because you can not differentiate between a line terminator at the end of the line and no line terminator.

That's what I meant by "with an additional check if you get a one-past-the-end result".  Basically, what I'm thinking is this (untested):

```rust
pub fn next_line_break(text: &str) -> Option<usize> {
    let byte_idx = lines::to_byte_idx(text, 1);
    if byte_idx == text.len() && !ends_with_line_break(text) {
        None
    } else {
        Some(byte_idx)
    }
}
```

(Note that `ends_with_line_break()` is already implemented, so you don't need to re-implement.)

> Correctly handling this would require checking for a line ending again which is really cumbersome and also slow

The check only happens when you reach the end of the text, so I suspect the amortized cost will be small.

> Furthermore the case of only looking for the first newline terminator can be easier optimized because it requires fewer checks

While that's true, I also think there's a balance to be struck between squeezing for performance and having a smaller code base with a smaller API surface area.  If it turns out it's a large performance boost, then it may well be worth it, and I'll certainly consider merging the PR.  But be prepared for a possible rejection as well.

> I would also like to SIMD accelerate the reverse case which is not possible to implement with any function is str_indicies at all

This, on the other hand, is something I would love to have in `str_indices`.  For the `chars` and `utf16` modules as well (though you don't have to tackle those too, just thinking out loud).  But I'd prefer that it be the generalized version, as in `fn to_byte_idx_rev(text: &str, line_idx_rev: usize)` rather than a narrow "prev" method, as per my comments above about the forward case.

Reply:
I have switched back to the `line_to_byte_idx(_, 1)`